q2_feature_table/tests/test_filter.py

```diff
@@ -340,6 +437,252 @@ def test_combine_frequency_filters(self):
+        metadata = qiime2.Metadata(df)
+        table = Table(np.array([[0, 1, 3], [1, 1, 2]]),
+                      ['O1', 'O2'],
+                      ['S1', 'S2', 'S3'])
+        actual = filter_samples(table,
+                                metadata=metadata,
+                                exclude_ids=True,
+                                max_frequency=4)
+        expected = Table(np.array([[1], [1]]),
+                         ['O1', 'O2'],
+                         ['S2'])
+        self.assertEqual(actual, expected)
+
+        # exclude one, max & min_frequency filter none
+        df = pd.DataFrame({'Subject': ['subject-1'],
+                           'SampleType': ['gut']},
+                          index=['S1'])
+        metadata = qiime2.Metadata(df)
+        table = Table(np.array([[0, 1, 3], [1, 1, 2]]),
+                      ['O1', 'O2'],
+                      ['S1', 'S2', 'S3'])
+        actual = filter_samples(table,
+                                metadata=metadata,
+                                exclude_ids=True,
+                                max_frequency=42,
+                                min_frequency=0)
+        expected = Table(np.array([[1, 3], [1, 2]]),
+                         ['O1', 'O2'],
+                         ['S2', 'S3'])
+        self.assertEqual(actual, expected)
+
+        # exclude one, min_frequency filter one,
+        # max_frequency filter one
+        df = pd.DataFrame({'Subject': ['subject-1'],
+                           'SampleType': ['gut']},
+                          index=['S1'])
+        metadata = qiime2.Metadata(df)
+        table = Table(np.array([[0, 1, 3], [1, 1, 2]]),
+                      ['O1', 'O2'],
+                      ['S1', 'S2', 'S3'])
+        actual = filter_samples(table,
+                                metadata=metadata,
+                                exclude_ids=True,
+                                max_frequency=4,
+                                min_frequency=3)
+        expected = Table(np.array([]), [], [])
+        self.assertEqual(actual, expected)
+
+        # where filter one -> exclude one,
+        # min_frequency filter one,
+        # max_frequency filter one
+        df = pd.DataFrame({'Subject': ['subject-1', 'subject-2'],
+                           'SampleType': ['gut', 'tongue']},
+                          index=['S1', 'S2'])
+        metadata = qiime2.Metadata(df)
+        table = Table(np.array([[0, 1, 3], [1, 1, 2]]),
+                      ['O1', 'O2'],
+                      ['S1', 'S2', 'S3'])
+        where = "Subject='subject-1'"
+        actual = filter_samples(table,
+                                metadata=metadata,
+                                exclude_ids=True,
+                                where=where,
+                                max_frequency=4,
+                                min_frequency=3)
+        expected = Table(np.array([]), [], [])
+        self.assertEqual(actual, expected)
+
```

Comment:
To be clear, you want me to demonstrate that if I exclude id `S3` and then pass a min filter that would only let S3 through, I should end up with nothin, since I excluded S3 and nothing else satisfies that filter, right?  That's what you'd like me to demonstrate, f or min and then max, that you can't basically "filter back in" something you just excluded by id?